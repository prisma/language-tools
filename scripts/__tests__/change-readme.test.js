
const { getNewReadMeContent } = require('./../change_readme.js')

describe('changeReadme', () => {

  it('it should with an insider release', () => {
    expect(
      getNewReadMeContent({
        releaseChannel: 'dev',
        npmVersion: '2.5.0-dev.4',
      })
    ).toMatchSnapshot()
  })

  it('it should work with a patch release', () => {
    expect(
        getNewReadMeContent({
            releaseChannel: 'patch-dev',
            npmVersion: '2.5.1-dev.1',
          })
    ).toMatchSnapshot()
  })

  it('it should work with a stable release', () => {
    expect(
        getNewReadMeContent({
            releaseChannel: 'latest',
            npmVersion: '2.4.0',
          })
    ).toMatchSnapshot()
  })
})
