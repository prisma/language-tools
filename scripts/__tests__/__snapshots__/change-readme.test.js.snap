// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeReadme it should with an insider release 1`] = `
"<h2 align=\\"center\\">Prisma VS Code Extension</h2>
<div align=\\"center\\">

![CI tests](https://img.shields.io/github/workflow/status/prisma/vscode/CI%20tests/master.svg?label=CI%20tests&logo=github)
![Check for Update + Automated Publish](https://img.shields.io/github/workflow/status/prisma/vscode/Check%20for%20Update%20%2B%20Automated%20Publish/master.svg?label=Check%20for%20Update%20%2B%20Automated%20Publish&logo=github)


</div>
<hr>

## Insider Build

This is the Insider Build of the [Prisma VSCode extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma). Most users should not use this version, but instead install
the regular version of the extension.This version of the extension may contain unstable prerelease code and is directly built from the master branch. 

**Only use this extension if you are also using the dev version of the CLI.**

Please note that you should not have the regular and insider version of the extension installed at the same time.

## Build information

This version is built from commit [undefined](undefined).
This is for Prisma CLI 2.5.0-dev.4.

## Contributing

Read more about [how to contribute to the Prisma VSCode extension](./packages/vscode/CONTRIBUTING.md)

## Telemetry

This extension collects telemetry data using [Azure Application Insights](https://www.npmjs.com/package/vscode-extension-telemetry) to help us better the usage of the extension.
The extension respects the \`telemetry.enableTelemetry\` setting in VSCode.

## Security

If you have a security issue to report, please contact us at [security@prisma.io](mailto:security@prisma.io?subject=[GitHub]%20Prisma%202%20Security%20Report%20VSCode)
"
`;

exports[`changeReadme it should work with a patch release 1`] = `
"<h2 align=\\"center\\">Prisma VS Code Extension</h2>
<div align=\\"center\\">

![CI tests](https://img.shields.io/github/workflow/status/prisma/vscode/CI%20tests/master.svg?label=CI%20tests&logo=github)
![Check for Update + Automated Publish](https://img.shields.io/github/workflow/status/prisma/vscode/Check%20for%20Update%20%2B%20Automated%20Publish/master.svg?label=Check%20for%20Update%20%2B%20Automated%20Publish&logo=github)


</div>
<hr>

## Insider Build

This is the Insider Build of the [Prisma VSCode extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma). Most users should not use this version, but instead install
the regular version of the extension.This version of the extension may contain unstable prerelease code and is directly built from the master branch. 

**Only use this extension if you are also using the dev version of the CLI.**

Please note that you should not have the regular and insider version of the extension installed at the same time.

## Build information

This version is built from commit [undefined](undefined).
This is for Prisma CLI 2.5.1-dev.1.

## Contributing

Read more about [how to contribute to the Prisma VSCode extension](./packages/vscode/CONTRIBUTING.md)

## Telemetry

This extension collects telemetry data using [Azure Application Insights](https://www.npmjs.com/package/vscode-extension-telemetry) to help us better the usage of the extension.
The extension respects the \`telemetry.enableTelemetry\` setting in VSCode.

## Security

If you have a security issue to report, please contact us at [security@prisma.io](mailto:security@prisma.io?subject=[GitHub]%20Prisma%202%20Security%20Report%20VSCode)
"
`;

exports[`changeReadme it should work with a stable release 1`] = `
"<h2 align=\\"center\\">Prisma VS Code Extension</h2>
<div align=\\"center\\">

[![Version](https://vsmarketplacebadge.apphb.com/version/prisma.Prisma.svg)](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma)
[![Installs](https://vsmarketplacebadge.apphb.com/installs/prisma.Prisma.svg)](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma)
[![Ratings](https://vsmarketplacebadge.apphb.com/rating/prisma.Prisma.svg)](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma)

</div>
<hr>
Adds syntax highlighting, formatting, jump-to-definition and linting for Prisma Schema files.

## Features

* Syntax highlighting
* Code Completion
    * Completion results appear for symbols as you type. 
    * You can trigger this manually with the \`Ctrl+Space\` shortcut.
* Documentation help
    * Information about the documentation of a completion result pops up as completion results are provided.
* Quick info on hover
    * Documentation Comments (\`///\`) of models and enums appear anywhere you hover over their usages.
* Go to Definition
    * Jump to or peek a model or enum's declaration.
* Formatting
    * Format code either manually or on save (if configured). 
    * *To automatically format on save, add the following to your \`settings.json\` file:*
        \`\`\`
        \\"editor.formatOnSave\\": true
        \`\`\`
* Rename
    * Rename models, enums, fields and enum values
    * Click into the model or enum, press \`F2\` and then type the new desired name and press \`Enter\`
    * All usages will be renamed
    * Automatically applies \`@map\` or \`@@map\` on the schema 
* Linting
    * Diagnostic tools are used to surface errors and warnings in your schema file as you type.
* Quick-fixes
    * Quickly fix typos in model and enum names
    * Create new models and enums with a single click


## Preview

<details>
  <summary>Syntax-Highlighting</summary>

![Preview Schema](https://i.imgur.com/W80iRwE.png)

</details>
<details>
  <summary>Quick-Fixes</summary>
  
![Quick-Fixes](https://media.giphy.com/media/iGGCF9JWOHx7SAKQlu/giphy.gif)
</details>

## Contributing

Read more about [how to contribute to the Prisma VSCode extension](./packages/vscode/CONTRIBUTING.md)

## Telemetry

This extension collects telemetry data using [Azure Application Insights](https://www.npmjs.com/package/vscode-extension-telemetry) to help us better the usage of the extension.
The extension respects the \`telemetry.enableTelemetry\` setting in VSCode.

## Build information

This is for Prisma CLI 2.4.0.

## Security

If you have a security issue to report, please contact us at [security@prisma.io](mailto:security@prisma.io?subject=[GitHub]%20Prisma%202%20Security%20Report%20VSCode)
"
`;
